### PR TITLE
Propagate errors during build to the Azure DevOps timeline

### PR DESCRIPTION
**Pull request by elinor-fung**
Propagate errors during build to the Azure DevOps timeline
- Use Arcade's msbuild script for building managed product and test projects. They include a logger when running under CI that outputs msbuild errors such that they appear in the timeline (and thus will be picked up by our telemetry)
- Prefix error messages to make them logging commands when running as an Azure DevOps task

This makes it so that the actual build/restore errors from managed projects will show up as errors in CI builds (and get classified as Code failures in telemetry). For native projects, the actual build errors are still not propagated, but we will at least have an error indicating that we failed to build the native components (instead of just Bash/Cmd exited with code 1). I also did not try to categorize our error messages - just made them propagate as errors instead of normal output.

### CODE REVIEW

File: build-test.cmd [hunk 1]
```diff
@@ -270,12 +276,13 @@ set __MsbuildWrn=/flp1:WarningsOnly;LogFile="%__BuildWrn%"
 set __MsbuildErr=/flp2:ErrorsOnly;LogFile="%__BuildErr%"
 set __Logging=!__MsbuildLog! !__MsbuildWrn! !__MsbuildErr!
 
-call "%__ProjectDir%\dotnet.cmd" msbuild /nologo /verbosity:minimal /clp:Summary /nodeReuse:false^
-  /l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log^
+REM Disable warnAsError - coreclr issue 19922
+powershell -NoProfile -ExecutionPolicy ByPass -NoLogo -Command "%__ProjectDir%\eng\common\msbuild.ps1" %__ArcadeScriptArgs%^
+  %__ProjectDir%\tests\build.proj -warnAsError:0 /t:BatchRestorePackages /nodeReuse:false^
+  '/l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log'^
```
Comment (jashook) [inline build-test.cmd]: While you are here do you mind removing the bin clash logger?

**Comment by AaronRobinsonMSFT** [comment]
cc @jeffschwMSFT

**Comment by sbomer** [comment]
Would you be able to add the prefix to `init-tools`, in particular https://github.com/dotnet/coreclr/blob/master/init-tools.sh#L176?

**Comment by elinor-fung** [comment]
@sbomer I'll just merge this first since all the builds passed already and create a separate PR handling error messages in `init-tools`

**Comment by sbomer** [comment]
I'm looking into what's left to remove buildtools so maybe it won't be necessary anyway - feel free to hold off

**Pull request merged**